nlp/2021_06_29_long_range_transformers/LongRangeTransformers.ipynb

```diff
@@ -0,0 +1,1940 @@
+    "\n",
+    "In recent research there has been an extensive study for improving the calculation of attention in transformer architectures. Mostly for improving their capacity to handle longer token sequences. 👊\n",
+    "\n",
+    "The attention calculation is known to be quadratic in compatation time with respect to the sequence length 👎. These recent advances, however, are able to perform attention calculation in linear time with respect to the sequence length. This allows us to scale the transformer architecture such that it can handle input sequences beyond the usual token length of 512. \n",
+    "\n",
+    "In this notebook, we compare traditional transformers with novel efficient transformers. We'll use roBERTa as a baseline to compare against LongFormer and BigBird.  \n",
+    "\n",
+    "Let's put these architectures to the test and see which one comes out on top 🏆!  \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "l8ymZTD_BSZS"
+   },
+   "source": [
+    "## 🛠️ Getting started: Install packages & download models\n",
+    "\n",
+    "The below cells will setup everything that is required to get started with keyword extraction:\n",
+    "\n",
+    "* Install python specific packages\n",
+    "* Import required packages"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "id": "eDI558ScBWpZ"
+   },
+   "outputs": [],
+   "source": [
+    "!pip install -q sklearn transformers datasets torch plotly"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "id": "x7x9s2l_VMLV"
+   },
+   "outputs": [],
+   "source": [
+    "import time \n",
+    "import sys \n",
+    "import json\n",
+    "import pandas as pd\n",
+    "from enum import Enum\n",
+    "import torch\n",
+    "\n",
+    "import plotly.express as px\n",
+    "import plotly.graph_objects as go\n",
+    "\n",
+    "\n",
+    "from sklearn.metrics import accuracy_score, precision_recall_fscore_support\n",
+    "from transformers import BigBirdTokenizerFast, BigBirdForSequenceClassification, RobertaTokenizer, RobertaForSequenceClassification, LongformerForSequenceClassification, TrainingArguments, Trainer, LongformerTokenizerFast\n",
+    "from datasets import load_dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "UOdkK5pXBSZT"
+   },
+   "source": [
+    "## 💾 Dataset & downstream task\n",
+    "\n",
+    "We will use the [IMDB](https://huggingface.co/datasets/imdb) movie review dataset for binary sentiment classification. In the paper publication of LongFormer and BigBird, both architectures were compared against RoBERTa with this exact dataset.\n",
+    "*   13.6 % of the reviews are larger than 512 wordpieces \n",
+    "*   evaluation was done on the entire dataset and the performance improvement was .4% and .8% compared to the RoBERTa architecture for LongFormer and BigBird respectively 💪.\n",
```

Comment:
- again, I would refrain from stating exact numbers in the text, as this might differ a tiny bit from run to run. 

- `evaluation was done on the entire dataset` > gives the impression that we evaluated on the training set > rephrase

Reply:
Those are the statistical results mentioned in the paper, maybe I should clarify this with, "The BigBird & LongFormer papers state the following: ", or leave it out? what do you think?